src/staking/phase1/index.ts

```diff
@@ -0,0 +1,349 @@
+  private stakerInfo: StakerInfo;
+
+  constructor(network: networks.Network, stakerInfo: StakerInfo) {
+    if (!isValidStakerInfo(stakerInfo, network)) {
+      throw new StakingError(
+        StakingErrorCode.INVALID_INPUT, "Invalid staker info",
+      );
+    }
+    this.network = network;
+    this.stakerInfo = stakerInfo;
+  }
+
+  /**
+   * Create a staking transaction for Phase 1 of the Babylon Staking protocol.
+   * 
+   * @param {Phase1Params} params - The staking parameters for Phase 1.
+   * @param {number} stakingAmountSat - The amount to stake in satoshis.
+   * @param {number} stakingTimeBlocks - The time to stake in blocks.
+   * @param {string} finalityProviderPublicKey - The finality provider's public key.
+   * @param {UTXO[]} inputUTXOs - The UTXOs to use as inputs for the staking 
+   * transaction.
+   * @param {number} feeRate - The fee rate for the transaction in satoshis per byte.
+   * 
+   * @returns {PsbtTransactionResult} - An object containing the unsigned psbt and fee
+   */
+  public createStakingTransaction = (
+    params: Phase1Params,
+    stakingAmountSat: number,
+    stakingTimeBlocks: number,
+    finalityProviderPublicKey: string,
+    inputUTXOs: UTXO[],
+    feeRate: number,
+  ): PsbtTransactionResult => {
+    const stakingTerm = getStakingTerm(params, stakingTimeBlocks);
+
+    validateStakingTxInputData(
+      stakingAmountSat,
+      stakingTerm,
+      params,
+      inputUTXOs,
+      feeRate,
+    );
+
+    const scripts = buildScripts(
+      params,
+      finalityProviderPublicKey,
+      stakingTerm,
+      this.stakerInfo.publicKeyHex,
+    );
+
+    // Create the staking transaction
+    try {
+      return stakingTransaction(
+        scripts,
+        stakingAmountSat,
+        this.stakerInfo.address,
+        inputUTXOs,
+        this.network,
+        feeRate,
+        isTaproot(this.stakerInfo.address, this.network) ? Buffer.from(this.stakerInfo.publicKeyHex, "hex") : undefined,
+        // `lockHeight` is exclusive of the provided value.
+        // For example, if a Bitcoin height of X is provided,
+        // the transaction will be included starting from height X+1.
+        // https://learnmeabitcoin.com/technical/transaction/locktime/
+        params.activationHeight - 1,
+      );
+    } catch (error: unknown) {
+      throw StakingError.fromUnknown(
+        error, StakingErrorCode.BUILD_TRANSACTION_FAILURE,
+        "Cannot build unsigned staking transaction",
+      );
+    }
+  };
+
+  /**
+   * Create an unbonding transaction for Phase 1 of the Babylon Staking protocol.
+   * 
+   * @param {Phase1Params} stakingParams - The staking parameters for Phase 1.
+   * @param {Phase1Delegation} delegation - The delegation to unbond.
+   * 
+   * @returns {Psbt} - The unsigned unbonding transaction
+   * 
+   * @throws {StakingError} - If the delegation is invalid or the transaction cannot be built
+   */
+  public createUnbondingTransaction = (
```

Comment:
The unbonding transaction is the same whether we are on the phase-1 or phase-2. In a future issue, we can track creating a base `Staking` class which is extended by this one to only add the functionality that is additional for phase-1. Same for the withdrawal methods.